iOSClient/Main/Collection Common/NCCollectionViewCommon.swift

```diff
@@ -1736,12 +1732,19 @@ extension NCCollectionViewCommon: UICollectionViewDataSource {
             footer.buttonIsHidden(true)
             footer.hideActivityIndicatorSection()
 
+
             if isSearchingMode {
                 if sections > 1 && section != sections - 1 {
                     footer.separatorIsHidden(false)
                 }
 
-                if isSearchingMode && isPaginated && metadatasCount > 0 && !isLastPage {
+                // If the number of entries(metadatas) is lower than the cursor, then there are no more entries.
+                // The blind spot in this is when the number of entries is the same as the cursor. If so, we don't have a way of knowing if there are no more entries.
+                // This is as good as it gets for determining last page without server-side flag.
+                let isLastPageTest = (metadatasCount < metadataForSection?.lastSearchResult?.cursor ?? 0) || metadataForSection?.lastSearchResult?.entries.isEmpty == true
+
```

Comment:
Important to note

Reply:
`isLastPageTest` was renamed to `isLastPage`, besides that ignore the Outdated tag